enhancements/hypershift/multi-arch-hypershift.md

```diff
@@ -0,0 +1,195 @@
+their perspective, and does not include the implementation details
+from the proposal.
+
+### User Stories
+
+#### Control Cluster (x86) with Non-x86 Compute Nodes
+As a cluster admistrator of a control cluster with HyperShift enabled, I would
+like to add my IBM Power, IBM zSystem, and/or ARM compute nodes to my new
+cluster so that I can run my architecture specific workloads without having to
+set up another control plane with additional hardware. I want this to work
+regardless of whether my additional nodes in a public cloud or not.
+
+- This means that whereever non-x86 instances that support OpenShift are
+ available (ARM on AWS, Azure, etc.; IBM Power on Power VS for IBM Cloud), I
+ should be able to add them to my cluster.
+- All arches should support non public cloud hardware as well.
+
+#### Control Cluster (ARM) with Any Arch Compute Nodes
+As a cluster admistrator of an ARM cluster in a public cloud environment, I
+would like to provision clusters so that I can benefit from having my
+orchestration pods running on cost-effective instances.
+
+- This has major potential cost-savings for managed services, and other types of
+  long-standing static infrastructure.
+
+#### Control Cluster (Non-x86) with Any Arch Compute Nodes
+As a cluster administrator of a non-x86 control cluster with HyperShift enabled,
+I would like easily scale out more clusters without having to dedicate hardware
+to each control plane. I would also like to be able to add
+control-plane-architecture agnostic compute nodes to my environment so that I
+can run operators or technologies that are only available on a subset of
+architectures in my environment.
+
+- All architectures benefit from HyperShift's scaling technology, but non-x86
+  clusters also benefit majorly from being able to add in x86 nodes for
+  dependencies that are only built for that architecture. This is especially
+  important when you consider all of the community and partner content that
+  is only available for x86.
+
+### Non-Goals
+
+This enchancment assumes that the control cluster where HyperShift is enabled is
+fully homogeneous. It does not explore the anything related to a heterogeneous
+control cluster, since that adds a third layer of architecture mismatch. Above
+we talk about the architecture of the control plane, and the architecture of the
+remote compute nodes. If you include heterogeneous clusters, we'd also have to
+talk about the architecture of the compute node of the control cluster where the
+pod-based control plane is running. This however, is the same discussion as one
+would have for any operator running on a heterogeneous cluster, and is thus best
+discussed and analyzed as part of the
+[Heterogeneous enhancement proposal](https://github.com/openshift/enhancements/pull/1014).
+
+## Proposal
+
+TODO
+
+### Workflow Description
+
+TODO
+
+
+### API Extensions
+
+N/a
+
+### Implementation Details/Notes/Constraints [optional]
+
+TODO
```

Comment:
Here's a first stab at some content for this section...

#### ARM

The capability to create an ARM NodePool on an x86 hosted control plane on AWS should be available for 4.12. The three ways of adding an ARM NodePool are through the HyperShift Create Cluster CLI, the HyperShift Create NodePool CLI, and through the HyperShift CRD. 

_HyperShift Create Cluster CLI_ - An `arch` flag has been added to the create cluster command specifying which architecture should be used for the default NodePool when the `--node-pool-replicas` flag is used. A manifest listed release image also needs to be specified in the `--release-image` flag of the command. 

Example: 
```
REGION=us-east-1
CLUSTER_NAME=example
BASE_DOMAIN=example.com
AWS_CREDS="$HOME/.aws/credentials"
PULL_SECRET="$HOME/pull-secret"
RELEASE_IMAGE="quay.io/openshift-release-dev/ocp-release@sha256:1a101ef5215da468cea8bd2eb47114e85b2b64a6b230d5882f845701f55d057f"
ARCH="aarch64"
​
hypershift create cluster aws \
--name $CLUSTER_NAME \
--node-pool-replicas=3 \
--base-domain $BASE_DOMAIN \
--pull-secret $PULL_SECRET \
--aws-creds $AWS_CREDS \
--region $REGION \
--release-image $RELEASE_IMAGE \
--arch $ARCH \
--generate-ssh
```
<br/>

_HyperShift Create NodePool CLI_  - An `arch` flag has been added to the create NodePool command specifying which architecture should be used for the default NodePool. As long as an x86 hosted cluster was created with a manifest listed image, ARM NodePools can be added to the hosted cluster as follows:
```
CLUSTER_NAME=example
NODE_POOLNAME=example-worker
ARCH="aarch64"
​
hypershift create nodepool aws \
--cluster-name $CLUSTER_NAME \
--name $NODE_POOLNAME \
--node-count=3 \
--arch $ARCH \
```
</br>

_HyperShift CRD_ - With the NodePool CRD, the `arch` field is located under the `schema.properties.spec` section.


Currently, the only valid values for `arch` are 'aarch64' and 'x86_64'. 'x86_64' is the default when `arch` isn't used.

The default AWS ARM instance type is `m6g.large`.